h/buildext.py

```diff
@@ -219,80 +236,78 @@ def build_chrome(args):
+            sentry_dsn = 'https:'.format(sentry_dsn)
 
     if webassets_env.url.startswith('chrome-extension:'):
-        build_extension_common(env, bundle_app=True)
+        build_extension_common(webassets_env, base_url, bundle_app=True)
         with codecs.open(content_dir + '/app.html', 'w', 'utf-8') as fp:
             data = client.render_app_html(
                 api_url=api_url,
-                base_url=base_url,
+                base_url=url_with_path(base_url),
 
                 # Google Analytics tracking is currently not enabled
                 # for the extension
                 ga_tracking_id=None,
-                sentry_dsn=sentry_dsn,
                 webassets_env=webassets_env,
-                websocket_url=websocket_url)
+                websocket_url=websocket_url,
+                sentry_dsn=sentry_dsn)
             fp.write(data)
     else:
-        build_extension_common(env)
+        build_extension_common(webassets_env, base_url)
 
     # Render the manifest.
     with codecs.open('build/chrome/manifest.json', 'w', 'utf-8') as fp:
-        data = chrome_manifest(env['request'])
+        script_url = urlparse.urlparse(webassets_env.url)
+        script_host_url = '{}://{}'.format(script_url.scheme,
+                                           script_url.netloc)
+        data = chrome_manifest(script_host_url)
         fp.write(data)
 
     # Write build settings to a JSON file
     with codecs.open('build/chrome/settings-data.js', 'w', 'utf-8') as fp:
+        settings = settings_dict(base_url, api_url, sentry_dsn)
         fp.write('window.EXTENSION_CONFIG = ' + json.dumps(settings))
 
 
 parser = argparse.ArgumentParser('hypothesis-buildext')
-
-parser.add_argument('config_uri', help='paster configuration URI')
-
 parser.add_argument('--debug',
                     action='store_true',
                     default=False,
                     help='create source maps to enable debugging in browser')
-
-subparsers = parser.add_subparsers(title='browser', dest='browser')
-subparsers.required = True
-
-parser_chrome = subparsers.add_parser(
-    'chrome',
-    help="build the Google Chrome extension",
-    description=textwrap.dedent(build_chrome.__doc__),
-    formatter_class=argparse.RawDescriptionHelpFormatter)
-parser_chrome.add_argument('--base',
-                           help='Base URL',
-                           default='http://localhost:5000',
-                           metavar='URL')
-parser_chrome.add_argument('--assets',
-                           help='A path (relative to base) or URL from which '
-                           'to load the static assets',
-                           default=None,
-                           metavar='PATH/URL')
+parser.add_argument('--sentry-dsn',
+                    default='',
+                    help='Specify the Sentry DSN for crash reporting',
+                    metavar='SENTRY_DSN')
+parser.add_argument(
+    '--service',
```

Comment:
Not sure why this argument is formatted completely differently from all the others. If it could not be that would be lovely.